src/commands/commit/list.ts

```diff
@@ -0,0 +1,116 @@
+import { Flags, ux } from "@oclif/core";
+import { AxiosResponse } from "axios";
+
+import * as ApiV1 from "@/lib/api-v1";
+import BaseCommand from "@/lib/base-command";
+import { formatDate } from "@/lib/helpers/date";
+import { merge } from "@/lib/helpers/object";
+import {
+  maybePromptPageAction,
+  pageFlags,
+  paramsForPageAction,
+} from "@/lib/helpers/page";
+import { withSpinner } from "@/lib/helpers/request";
+import { formatCommitAuthor } from "@/lib/marshal/commit";
+
+export default class CommitList extends BaseCommand<typeof CommitList> {
+  static summary = "Display all commits in an environment";
+
+  static flags = {
+    environment: Flags.string({
+      default: "development",
+      summary: "The environment to use.",
+    }),
+    promoted: Flags.boolean({
+      summary:
+        "Show only promoted or unpromoted changes between the given environment and the subsequent environment.",
+      allowNo: true,
+    }),
+    ...pageFlags,
+  };
+
+  static enableJsonFlag = true;
+
+  async run(): Promise<ApiV1.ListCommitResp | void> {
+    const resp = await this.request();
+
+    const { flags } = this.props;
+    if (flags.json) return resp.data;
+
+    this.render(resp.data);
+  }
+
+  async request(pageParams = {}): Promise<AxiosResponse<ApiV1.ListCommitResp>> {
+    const props = merge(this.props, { flags: { ...pageParams } });
+
+    return withSpinner<ApiV1.ListCommitResp>(() =>
+      this.apiV1.listCommits(props),
+    );
+  }
+
+  async render(data: ApiV1.ListCommitResp): Promise<void> {
+    const { entries } = data;
+    const { environment: env, promoted } = this.props.flags;
+
+    let qualifier = "";
+
+    if (promoted) {
```

Comment:
```suggestion
    if (promoted === true) {
```
Total nit, feel free to ignore - but for the symmetry and explicitness.